tests/test_datasets/test_hierarchical_dataset.py

```diff
@@ -0,0 +1,169 @@
+import pandas as pd
+import pytest
+
+from etna.datasets.hierarchical_structure import HierarchicalStructure
+from etna.datasets.tsdataset import TSDataset
+
+
+@pytest.fixture
+def hierarchical_structure():
+    hs = HierarchicalStructure(
+        level_structure={"total": ["X", "Y"], "X": ["a", "b"], "Y": ["c", "d"]},
+        level_names=["total", "market", "product"],
+    )
+    return hs
+
+
+@pytest.fixture
+def diff_level_segments_df():
```

Comment:
Part `diff` may be misleading. Usually `diff` means difference but here, as I understand, it means different.